src/julia_gcext.h

```diff
@@ -49,6 +49,12 @@ JL_DLLEXPORT jl_datatype_t *jl_new_foreign_type(
         int haspointers,
         int large);
 
+
```

Comment:
Actually, could we insert this? That would make it a tad easier to write code that works with both the old and the new model:
```suggestion

#define HAVE_JL_REINIT_FOREIGN_TYPE
```